hapi-proto/src/main/proto/ScheduleService.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2020 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+option java_package = "com.hederahashgraph.api.proto.java";
+option java_multiple_files = true;
+
+import "Query.proto";
+import "Response.proto";
+import "TransactionResponse.proto";
+import "Transaction.proto";
+
+/* Transactions and queries for the Schedule Service */
+service ScheduleService {
+  // Creates a new Schedule by submitting the transaction
+  rpc createSchedule (Transaction) returns (TransactionResponse);
```

Comment:
I think not having a `updateSchedule` and attempting to have two different flows for create whether it exists or not is a mistake. We should separate the two flows and have an `updateSchedule` operation that allows updating fields except for the `transactionBody`. Could even allow updating of the `transactionBody` if nobody has signed except the creator account.

Reply:
Having 2 different flows for creating (whether it exists or not) is adding more complexity for sure and you are right to think that way. However, we have a very important functional requirement that we want to support and its the following:

- We want for clients to be able to create scheduled transactions without the need for them to communicate with each other. F.e everyone wants to execute the same transaction so they submit it. This results in 1 scheduled tx creation and 5 signatures being appended. There is no need for the creator of the tx to provide the ID of the scheduled TX to the others so that they can append their signatures.

If we make the creation flow to support only creation (without append), we are forcing participants to have:
1) Leader (who gets to create the scheduled tx)
2) Communication between parties (everyone should get the scheduled tx id from the "leader")

If the setup requires a leader and communication between the parties they don't need to use the scheduled TX at all. They can set up a threshold account and provide their signatures to the leader.

Reply:
Good point. I think there might be other solutions to consider keeping in mind that requirement:

- Provide a query to search for scheduled transactions by transaction body (or its hash) and the payer account (or whatever is its unique composite key to switch between update/create flow internally) which clients can use if they get "schedule already exists"
- Return existing schedule ID on failed schedule create receipt or in error message. Would avoid the extra query transaction
- Provide an ScheduleUpdate that can take either schedule ID or transaction body plus payer to uniquely identify

Reply:
I think the strongest argument for separating create vs update is that clients will get unexpected results depending if they are the first to create or they do a create that actually updates. This is because the update will ignore every field in the schedule create except for the `sigMap`.

For example, if Alice and Bob both submit identical `ScheduleCreate` transactions except for `adminKey` and Alice succeeds one nanosecond before Bob, Bob will be surprised that his `adminKey` is not set. Even worse, if Alice and Bob send identical `ScheduleCreate` transactions except that Alice requires 2 signatures but Bob requires 3, then Bob will be signing a 2 of 2 instead of the expected 3 of 3. Though I might be misunderstanding which `transactionBody` and `sigMap` are used as it's quite unclear to me since they're all named the same.

Reply:
Discussed and we won't be adding an update as part of this MVP. It's either ScheduleCreate with an actual create or an implicit sign. As such, it won't change the protobuf so we can defer.